### DISCUSSION TITLE
Ubuntu & SSH keys

### DISCUSSION THREAD

**Discussion by JamesArthurHarrison** [remotemobprogramming/mob#182] [Q&A]
Hi,

A month ago I asked for help with mob.sh and you all did a wonderful job of allowing mob.sh to connect to .gitignore files. That has now lead me to another potential issue... I'm not sure it isn't user error my end so I thought I'd start this as a discussion and then I can make it an issue if it is issue worthy!

So now when I use `mob start` I get the following error:
```09:55 $ mob start
 ERROR git fetch origin --prune
 ERROR git@gitlab.com: Permission denied (publickey,keyboard-interactive).
 ERROR fatal: Could not read from remote repository.
 ERROR 
 ERROR Please make sure you have the correct access rights
 ERROR and the repository exists.
 ERROR exit status 128
 ```

I've run `sudo snap connect mob-sh:ssh-keys`. I'm assuming this is linked to the "Know issue; SSH-agent not working"? We use a Yubikeys to connect to the repo - is this the cause of the issue?

Either way, is there something that can be done to enable us to use mob.sh on Ubuntu?

**Comment by simonharrer** [comment]
@nikolashermann can you help?

**Reply by nikolashermann** [reply]
Adding a plug for ssh-agent in a snap is not possible yet: https://forum.snapcraft.io/t/ssh-agent-plug-request/1250/4
I will test shipping an agent inside the snap.
@JamesArthurHarrison Could you provide some more information about how the yubikey-authentication is implemented? Is it this: https://developers.yubico.com/PGP/SSH_authentication/ ?

**Comment by simonharrer** [comment]
> Either way, is there something that can be done to enable us to use mob.sh on Ubuntu?

Just install it with `curl -sL install.mob.sh | sh`.

**Reply by JamesArthurHarrison** [reply]
I'll experiment with that now, thanks. Seems like there is some `snap` cleanup to be done before I can get `curl` working - I'll report back

**Reply by JamesArthurHarrison** [reply]
Well I've got that running now, but it would still be ideal to see if there is a solution for getting the SSH agent working via snap. As a team we would prefer to install it that way (Installing, removing & updating would them be much easier to handle via our own puppet script).

Also installing via `curl` came up with a "your system doesn't have a `say` command, thus there will be no sound` style error. any ideas how to get round this @simonharrer? It's not a big deal, but the timer being audible would be great. (moo worked via snap, and doesn't now...but mobbing does so that's better!)

**Reply by simonharrer** [reply]
You have to set up the sound output yourself. The install script should point you to the readme how to do that: https://mob.sh/#linux-timer

**Reply by JamesArthurHarrison** [reply]
Ah, hadn't read that ages ago & forgot when using `snap` & `mob moo` worked. Thanks again

**Reply by JamesArthurHarrison** [reply]
It took me long enough - but i worked it out. comment removed

**Reply by simonharrer** [reply]
Anything we could make to make that manual setup process easier?

**Reply by JamesArthurHarrison** [reply]
Yes!

Sorry this has taken so long, I've not had time to look at this again till today. It seems trivial but adding a line mentioning setting the mode of the script to 775 would be great (or equivalent). Somehow we kept not quite getting it to work, and I think persistently having the wrong mode was the cause of this! Anyway not it's all good - I've got a lovely British "Moo" set up now! 

Thanks for your help!

**Comment by alex-nicholas** [comment]
I'm having this same issue on Ubuntu v22.04.2

I can use git to directly without issues, but if I try to use `mob s 15` i get :

```
ERROR git fetch origin --prune
ERROR Failed to add the ECDSA host key for IP address '20.248.137.48' to the list of known hosts (<known_hosts  directory>).
ERROR git@github.com: Permission denied (publickey).
ERROR fatal: Could not read from remote repository.
ERROR 
ERROR Please make sure you have the correct access rights
ERROR and the repository exists.
ERROR exit status 128
```

**Reply by alex-nicholas** [reply]
So I got it to work by doing `sudo mob s 15`, it failed in the same way, but asked me to add the fingerprint to my known_hosts.
After that, I am able to run `mob s` and it work as expected.
